router/connection_maker.go

```diff
@@ -52,6 +52,13 @@ func (cm *ConnectionMaker) InitiateConnection(address string) {
 	}
 }
 
+func (cm *ConnectionMaker) ForgetConnection(address string) {
+	cm.actionChan <- func() bool {
+		delete(cm.cmdLineAddress, NormalisePeerAddr(address))
+		return true
```

Comment:
Yes I think you're right - the extant target for this address will get deleted either when the timer fires or ConnectionTerminated gets called. Will push an update.